docs/modules/candid-guide/index.adoc

```diff
@@ -289,23 +320,56 @@ If you want to create a new Candid implementation, this document will not suffic
 
-[#reference]
 == Reference
 
-This section lists all the types supported by Candid, including their type syntax, the syntax for the textual representation, their upgrading rules and the corresponding types in Rust, Motoko and Javascript.
+This section provides links to Candid tools and libraries for currently supported host languages and detailed reference information about Candid supported types.
+
+If you are looking for more information before you start working with Candid, check out the following related resources:
+
+* link:https://www.youtube.com/watch?v=O2KaWRtsqHg[How Candid provides a common language for application interfaces (video)]
+
+=== Tools and libraries
+
+This section provides links to Candid tools and libraries for currently-supported host languages.
+
+* The Motoko language has built-in support for Candid and the compiler uses it implicitly.
+
+* To create Candid services or clients in Rust, you can use the https://crates.io/crates/candid[Candid crate].
+
+* The https://github.com/dfinity/agent-js/[Internet Computer agent library for JavaScript] includes support for Candid.
+
+* Community-supported libraries for Candid also exist for https://hackage.haskell.org/package/candid[Haskell], https://github.com/chenyan2002/ic-elm/[Elm] and https://github.com/seniorjoinu/candid-kt[Kotlin].
+
+* The https://github.com/dfinity/candid/tree/master/tools/ui[Candid UI] creates a Web interface from the Candid interface of any canister.
+
+=== Supported types
+
+This section lists all the types supported by Candid.
+For each type, the reference includes the following information:
+
+* Type syntax and the syntax for the textual representation of the type.
+* Upgrade rules for the type.
+* Corresponding types in Rust, Motoko and Javascript.
+
+== Subtypes and supertypes
 
-TIP: Subtypes are the types you can change your method _results_ to, and supertypes are the types that you can change your method _arguments_ to.
+Subtypes are the types you can change your method _results_ to.
+Supertypes are the types that you can change your method _arguments_ to.
 
-We only list the particular subtypes and supertypes in each section, and do not repeat every time that that the type `empty` is a subtype of any type, and that the types `reserved` and `opt t` are supertypes of any types. See the sections on these types for more details.
+You should note that this reference only lists the specific subtypes and supertypes that are relevant for each type. 
+It does not repeat common information about subtypes and supertypes that can apply to any type. 
+For example, the reference does not list `+empty+` as a subtype because it can be a subtype of any other type.
+Similarly, the types `+reserved+` and `+opt t+` are not listed as supertypes of specific types because they are supertypes of any type. 
+The `+empty+`, `+reserved+`, and `+opt t+` types are discussed separately, independent of other types.
 
 [#type-text]
-=== Type text
+== Type text
```

Comment:
This and all the following sections are part of the language reference, so should probably remain to be subsections of Reference.